pkg/pgmodel/metrics/database/database.go

```diff
@@ -0,0 +1,112 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+type Engine interface {
+	Start()
+	Stop()
+}
+
+type metricsEngineImpl struct {
+	conn    pgxconn.PgxConn
+	metrics []metricQueryWrap
+	stop    chan struct{}
+}
+
+// NewEngine creates an engine that performs database metrics evaluation every evalInterval.
+// Make sure to call this only when the database is TimescaleDB. Plain Postgres
+// will cause evaluation errors.
+func NewEngine(conn pgxconn.PgxConn) *metricsEngineImpl {
+	return &metricsEngineImpl{
+		conn:    conn,
+		metrics: metrics,
+		stop:    make(chan struct{}),
+	}
+}
+
+func (e *metricsEngineImpl) register() {
+	prometheus.MustRegister(getMetrics(e.metrics)...)
+}
+
+func (e *metricsEngineImpl) unregister() {
+	prometheus.MustRegister(getMetrics(e.metrics)...)
+}
+
+func getMetrics(m []metricQueryWrap) []prometheus.Collector {
+	var metrics []prometheus.Collector
+	for i := range m {
+		metrics = append(metrics, m[i].metric)
+	}
+	return metrics
+}
+
+const sqlMetricsEvalInterval = time.Second * 3
```

Comment:
Is this correct?

Reply:
Thanks for pointing out. This should be 3 mins, I was debugging at 3 seconds and pushed it 😓

Reply:
Should this have a back-off mechanism? These metrics shouldn't add to the DB load for systems that are being stressed.

If its too much, we could do it in a follow-up PR.

Reply:
adding discussion here, so that it's easier to track, for backoff, something like:

```
 wihch means the interval will no longer be 3 mins, rather > 3 mins, till a range, like upto 6 mins
```